posts/app.py

```diff
@@ -53,6 +53,49 @@ def get_all_posts_for_event(event_id):
     """Get all posts matching the event with the specified ID."""
 
 
+def upload_new_post_to_db(post, collection):
+    """Uploads a new post to the db collection.
+
+    Assumes the event matching the post's `event_id` and the user matching
+        the post's `author_id` both exist. I.e. Caller should check this.
```

Comment:
Is it important that both exist? What happens if they do not?

It seems like this function will behave fine if they don't exist?

Reply:
It's important that they both exist so that the front end UI is able to query all posts for a given event ID and see the users that posted each post. Right now, this function behaves fine if they don't exist which is why I added this comment.

Reply:
One part of microservice design is that you don't want to have one domain (microservice) worry too much about constraints in another domain (microservice). So in this case, it would be the frontend that managed the concern about handling missing events and/or authors.

Reply:
Ah okay, done in https://github.com/knative-portability/large-events/pull/16/commits/775e718d10cb96e0abf66da99e9c2b80c7f2882d.